src/clients/api/queries/getXvsReward/getVTokenData.ts

```diff
@@ -0,0 +1,65 @@
+import Web3 from 'web3';
+import BigNumber from 'bignumber.js';
+
+import { getVTokenContract } from 'clients/contracts';
+import { Comptroller } from 'types/contracts';
+import { VTokenId } from 'types';
+
+export interface IGetVTokenDataInput {
+  tokenId: VTokenId;
+  tokenAddress: string;
+  accountAddress: string;
+  comptrollerContract: Comptroller;
+  web3: Web3;
+}
+
+export interface IGetVTokenDataOutput {
+  supplyStateIndex: string;
+  borrowStateIndex: string;
+  borrowBalanceStoredWei: BigNumber;
+  borrowIndex: string;
+  userBorrowIndex: string;
+  userSupplyIndex: string;
+  userBalanceWei: BigNumber;
+}
+
+export const getVTokenData = async ({
+  tokenId,
+  tokenAddress,
+  accountAddress,
+  comptrollerContract,
+  web3,
+}: IGetVTokenDataInput) => {
+  const vTokenContract = getVTokenContract(tokenId, web3);
+
+  const [
+    supplyState,
+    borrowState,
+    userBorrowIndex,
+    userSupplyIndex,
+    tokenBorrowIndex,
+    userBorrowBalanceStoredWei,
+    userBalanceWei,
+  ] = await Promise.all([
+    comptrollerContract.methods.venusSupplyState(tokenAddress).call(),
```

Comment:
We've been setting up every contract call as a separate queries and then grouping them together in a hook